app/src/main/cpp/skyline/vfs/android_asset_backing.h

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2021 Skyline Team and Contributors (https://github.com/skyline-emu/)
+
+#pragma once
+
+#include "backing.h"
+
+struct AAsset;
+
+namespace skyline::vfs {
+    /**
+     * @brief The AndroidAssetBacking class provides the backing abstractions for the AAsset Android API
+     * @note This is NOT thread safe NOR should it be shared across threads.
+     * @note This will take ownership of the backing asset passed into it
+     */
+    class AndroidAssetBacking : public Backing {
+      private:
+        AAsset *backing; //!< The backing AAsset object we abstract
```

Comment:
I recommend renaming this to prevent confusion with the backing abstraction in itself, it took me a minute to realize what this was
```suggestion
        AAsset *asset; //!< The underlying NDK AAsset object
```